objc/objc_runtime_darwin.go

```diff
@@ -152,17 +155,53 @@ func (c Class) AddMethod(name SEL, imp _IMP, types string) bool {
 	return byte(ret) != 0
 }
 
-// Register registers a class that was allocated using objc_allocateClassPair.
+// AddIvar adds a new instance variable to a class.
+// It may only be called after AllocateClassPair and before Register.
+// Adding an instance variable to an existing class is not supported.
+// The class must not be a metaclass. Adding an instance variable to a metaclass is not supported.
+// The instance variable's minimum alignment in bytes is 1<<align. The minimum alignment of an
+// instance variable depends on the ivar's type and the machine architecture.
+// For variables of any pointer type, pass math.Log2(unsafe.Alignof(type)).
```

Comment:
My understanding is that we cannot have a utility function to calculate the size due to `unsafe.Alignof` like this

```go
func IvarSize(value interface{}) uint8 {
    return uint8(math.Log2(unsafe.Alignof(value)))
}
```

Instead, can we have this utility function?

```go
func IvarSizeFromAlign(align uintptr) uint8 {
    return uint8(math.Log2(align))
}
```

Reply:
Another option is to calcuate `math.Log2` in `AddIvar` so that the user doesn't have to care.

Reply:
Well according to the docs `reflect.TypeOf(x).Align()` is equal to `unsafe.Alignof`. So it would be possible to do what you first suggested. Although I think it should one called `IvarAlignment` instead of `IvarSize` since it is calculating the alignment.

In your second comment are you referring to taking just unsafe.Alignof for the alignment parameter and then do math.Log2 inside AddIvar?

Perhaps, we make the signature `AddIvar(name string, ty interface{}, types string) bool` and then inside AddIvar calculate the size and alignment using reflect.

Reply:
> In your second comment are you referring to taking just unsafe.Alignof for the alignment parameter and then do math.Log2 inside AddIvar?

Yes.

> Perhaps, we make the signature AddIvar(name string, ty interface{}, types string) bool and then inside AddIvar calculate the size and alignment using reflect.

This seems the best idea so far. What do you think?

Reply:
I do like it. It makes using it a lot easier and makes it harder to use improperly.